Fail *.test.py tests in the case of catching server start errors

https://github.com/tarantool/tarantool/runs/5703424705?check_suite_focus=true

```
...

[054] replication-py/init_storage.test.py                             
[054] [Instance "replica"] Failed to start within 90 seconds
[054] 
[054] Last 15 lines of Tarantool Log file [Instance "replica"][/build/usr/src/debug/tarantool-2.10.0~beta2.209.dev/test/var/054_replication-py/replica.log]:
[054] 2022-03-26 13:18:53.207 [35962] main/103/replica I> synchronizing with 1 replicas
[054] 2022-03-26 13:18:53.207 [35962] main/112/applier/localhost:10780 I> subscribed
[054] 2022-03-26 13:18:53.207 [35962] main/112/applier/localhost:10780 I> remote vclock {1: 13} local vclock {1: 13}
[054] 2022-03-26 13:18:53.207 [35962] main/112/applier/localhost:10780 I> RAFT: message {term: 1, state: follower} from 1
[054] 2022-03-26 13:18:53.208 [35962] main/112/applier/localhost:10780 I> leaving orphan mode
[054] 2022-03-26 13:18:53.208 [35962] main/103/replica I> replica set sync complete
[054] 2022-03-26 13:18:53.208 [35962] main/103/replica I> leaving orphan mode
[054] 2022-03-26 13:18:53.208 [35962] main/103/replica I> set 'log_level' configuration option to 5
[054] 2022-03-26 13:18:53.208 [35962] main/107/checkpoint_daemon I> scheduled next checkpoint for Sat Mar 26 14:21:09 2022
[054] 2022-03-26 13:18:53.208 [35962] main/103/replica I> set 'replication_timeout' configuration option to 0.1
[054] 2022-03-26 13:18:53.208 [35962] main/103/replica I> set 'memtx_memory' configuration option to 107374182
[054] 2022-03-26 13:18:53.208 [35962] main/103/replica I> set 'listen' configuration option to "localhost:11525"
[054] 2022-03-26 13:18:53.208 [35962] main/103/replica I> set 'log_format' configuration option to "plain"
[054] 2022-03-26 13:18:53.208 [35962] main/103/replica I> set 'replication' configuration option to ["localhost:10780"]
[054] 2022-03-26 13:18:53.208 [35962] main I> entering the event loop
[054] 
[054] Worker "054_replication-py" received the following error; stopping...
[054] Traceback (most recent call last):
[054]   File "/build/usr/src/debug/tarantool-2.10.0~beta2.209.dev/test-run/lib/worker.py", line 319, in run_task
[054]     short_status, duration = self.suite.run_test(
[054]   File "/build/usr/src/debug/tarantool-2.10.0~beta2.209.dev/test-run/lib/test_suite.py", line 278, in run_test
[054]     short_status = test.run(server)
[054]   File "/build/usr/src/debug/tarantool-2.10.0~beta2.209.dev/test-run/lib/test.py", line 195, in run
[054]     self.execute(server)
[054]   File "/build/usr/src/debug/tarantool-2.10.0~beta2.209.dev/test-run/lib/tarantool_server.py", line 404, in execute
[054]     exec(code, new_globals)
[054]   File "replication-py/init_storage.test.py", line 26, in <module>
[054]     replica.deploy()
[054]   File "/build/usr/src/debug/tarantool-2.10.0~beta2.209.dev/test-run/lib/tarantool_server.py", line 761, in deploy
[054]     self.start(silent=silent, **kwargs)
[054]   File "/build/usr/src/debug/tarantool-2.10.0~beta2.209.dev/test-run/lib/tarantool_server.py", line 888, in start
[054]     self.wait_until_started(wait_load, deadline)
[054]   File "/build/usr/src/debug/tarantool-2.10.0~beta2.209.dev/test-run/lib/tarantool_server.py", line 1120, in wait_until_started
[054]     self.wait_load(deadline)
[054]   File "/build/usr/src/debug/tarantool-2.10.0~beta2.209.dev/test-run/lib/tarantool_server.py", line 1105, in wait_load
[054]     raise TarantoolStartError(

...

Statistics:
* pass: 1316
* disabled: 159
[Internal test-run error] The following tasks were dispatched to some worker task queue, but were not reported as done (does not matters success or fail):
- [replication-py/init_storage.test.py, null]
make[4]: *** [test/CMakeFiles/test-force.dir/build.make:73: test/CMakeFiles/test-force] Error 4
make[4]: Leaving directory '/build/usr/src/debug/tarantool-2.10.0~beta2.209.dev'
make[3]: *** [CMakeFiles/Makefile2:6039: test/CMakeFiles/test-force.dir/all] Error 2
make[3]: Leaving directory '/build/usr/src/debug/tarantool-2.10.0~beta2.209.dev'
make[2]: *** [CMakeFiles/Makefile2:6046: test/CMakeFiles/test-force.dir/rule] Error 2
make[2]: Leaving directory '/build/usr/src/debug/tarantool-2.10.0~beta2.209.dev'
make[1]: *** [Makefile:2285: test-force] Error 2
make[1]: Leaving directory '/build/usr/src/debug/tarantool-2.10.0~beta2.209.dev'
error: Bad exit status from /var/tmp/rpm-tmp.zGzXiD (%check)
```

As we can see, the ` replication-py/init_storage.test.py` test failed because the tarantool server was unable to start for some reason, but the test is not considered as a failed test. As a result of that, log artifacts are not gathered by test-run.

--- a/lib/tarantool_server.py
+++ b/lib/tarantool_server.py
@@ -459,11 +459,14 @@ class TarantoolLog(object):
                     return pos
 
     def seek_wait(self, msg, proc=None, name=None, deadline=None):
+        timeout = Options().args.server_start_timeout
         while not deadline or time.time() < deadline:
             if os.path.exists(self.path):
                 break
             gevent.sleep(0.001)
         else:
+            color_stdout("\nFailed to locate {} logfile within {} "
+                         "seconds\n".format(self.path, timeout), schema='error')
             return False
 
         with open(self.path, 'r') as f:
@@ -471,7 +474,10 @@ class TarantoolLog(object):
             cur_pos = self.log_begin
             while not deadline or time.time() < deadline:
                 if not (proc is None):
+                    # proc.poll() returns None if the process is working. When
+                    # the process completed, it returns the process exit code.
                     if not (proc.poll() is None):
+                        # Raise an error since the process completed.
                         raise TarantoolStartError(name)
                 log_str = f.readline()
                 if not log_str:
@@ -481,6 +487,11 @@ class TarantoolLog(object):
                 if re.findall(msg, log_str):
                     return True
                 cur_pos = f.tell()
+
+        color_stdout("\nFailed to find '{}' pattern in {} logfile within {} "
+                     "seconds\n".format(msg, self.path, timeout),
+                     schema='error')
+
         return False
 
 

--- a/lib/tarantool_server.py
+++ b/lib/tarantool_server.py
@@ -402,7 +402,13 @@ class PythonTest(Test):
         new_globals = dict(locals(), test_run_current_test=self, **server.__dict__)
         with open(self.name) as f:
             code = compile(f.read(), self.name, 'exec')
+
+        try:
             exec(code, new_globals)
+        except TarantoolStartError:
+            # fail tests in the case of catching server start errors
+            raise TestExecutionError
+
         # crash was detected (possibly on non-default server)
         if server.current_test.is_crash_reported:
             raise TestExecutionError

--- a/lib/worker.py
+++ b/lib/worker.py
@@ -266,9 +266,9 @@ class Worker:
                          'the tasks will be ignored...\n' % self.name,
                          schema='error')
             color_stdout("The raised exception is '%s' of type '%s'.\n"
-                         % (str(e), str(type(e))), schema='error')
+                         % (str(e).strip(), str(type(e))), schema='error')
             color_stdout('Worker "%s" received the following error:\n'
-                         % self.name + traceback.format_exc() + '\n',
+                         % self.name + traceback.format_exc().strip() + '\n',
                          schema='error')
             self.stop_server(cleanup=False)
 

